bash.sh

```diff
@@ -53,14 +53,34 @@ array_filter() {
 #
 # Join array elements with a string.
 #
-# $1: String separator
-# $@: Array elements
+# $1: String default, used in case array to join is empty
+# $2: String separator, has to be a single element
```

Comment:
Since we're making `array_join` better, why don't we improve it to allow any string, rather than just a single character?

I can imagine this is useful, e.g. say you want to create a human-readable CSV, so you want to join by ", " (comma-space).